exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -158,109 +171,213 @@ func (e *Exchange) Setup(exch *config.Exchange) error {
 		return err
 	}
 
-	wsRunningURL, err := e.API.Endpoints.GetURL(exchange.WebsocketSpot)
+	err = e.Websocket.Setup(&websocket.ManagerSetup{
+		ExchangeConfig: exch,
+		FillsFeed:      e.Features.Enabled.FillsFeed,
+		TradeFeed:      e.Features.Enabled.TradeFeed,
+		Features:       &e.Features.Supports.WebsocketCapabilities,
+		OrderbookBufferConfig: buffer.Config{
+			SortBuffer:            true,
+			SortBufferByUpdateIDs: true,
+		},
+		UseMultiConnectionManagement: true,
+	})
 	if err != nil {
 		return err
 	}
-
-	err = e.Websocket.Setup(&websocket.ManagerSetup{
-		ExchangeConfig:        exch,
-		DefaultURL:            poloniexWebsocketAddress,
-		RunningURL:            wsRunningURL,
+	err = e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		URL:                   poloniexWebsocketAddress,
+		RateLimit:             request.NewWeightedRateLimitByDuration(500 * time.Millisecond),
+		Subscriber:            e.Subscribe,
+		Unsubscriber:          e.Unsubscribe,
+		GenerateSubscriptions: func() (subscription.List, error) { return e.GenerateDefaultSubscriptions(false) },
+		Handler:               e.wsHandleData,
 		Connector:             e.WsConnect,
+		MessageFilter:         connSpotPublic,
+	})
+	if err != nil {
+		return err
+	}
+	err = e.Websocket.SetupNewConnection(&websocket.ConnectionSetup{
+		ResponseCheckTimeout:  exch.WebsocketResponseCheckTimeout,
+		ResponseMaxLimit:      exch.WebsocketResponseMaxLimit,
+		URL:                   poloniexPrivateWebsocketAddress,
```

Comment:
above and below so that it correctly applies to config and can be overridden if required. 
```go

wsSpotPrivate, err := e.API.Endpoints.GetURL(exchange.SpotPrivate)
	if err != nil {
		return err
	}

```

e.API.Endpoints.SetDefaultEndpoints needs to be updated